src/main/java/sg/ncl/MainController.java

```diff
@@ -725,30 +714,30 @@ public String signup2(Model model, HttpServletRequest request) {
         Map<String, ?> inputFlashMap = RequestContextUtils.getInputFlashMap(request);
         if (inputFlashMap != null) {
             log.debug((String) inputFlashMap.get(MESSAGE));
-            model.addAttribute("signUpMergedForm", (SignUpMergedForm) inputFlashMap.get("signUpMergedForm"));
+            model.addAttribute(SIGNUP_MERGED_FORM, (SignUpMergedForm) inputFlashMap.get(SIGNUP_MERGED_FORM));
         } else {
             log.debug("InputFlashMap is null");
-            model.addAttribute("signUpMergedForm", new SignUpMergedForm());
+            model.addAttribute(SIGNUP_MERGED_FORM, new SignUpMergedForm());
         }
-        return "signup2";
+        return SIGNUP_PAGE;
     }
 
     @RequestMapping(value = "/signup2", method = RequestMethod.POST)
     public String validateDetails(
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 32 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)